src/languages/swift.js

```diff
@@ -283,6 +288,143 @@ export default function(hljs) {
   };
   TYPE.contains.push(GENERIC_ARGUMENTS);
 
+  // https://docs.swift.org/swift-book/ReferenceManual/Expressions.html#ID552
+  // Prevents element names from being highlighted as keywords.
+  const TUPLE_ELEMENT_NAME = {
+    begin: lookahead(concat(Swift.identifier, /\s*:/)),
+    end: lookahead(/:/),
+    keywords: "_",
+    relevance: 0
+  };
+  // Matches tuples as well as the parameter list of a function type.
+  const TUPLE = {
+    begin: /\(/,
+    end: /\)/,
+    relevance: 0,
+    keywords: KEYWORDS,
+    contains: [
+      'self',
+      TUPLE_ELEMENT_NAME,
+      ...COMMENTS,
+      ...KEYWORD_MODES,
+      ...BUILT_INS,
+      ...OPERATORS,
+      NUMBER,
+      STRING,
+      ...IDENTIFIERS,
+      ...ATTRIBUTES,
+      TYPE
+    ]
+  };
+
+  // https://docs.swift.org/swift-book/ReferenceManual/Declarations.html#ID362
+  // Matches both the keyword func and the function title.
+  // Grouping these lets us differentiate between the operator function <
+  // and the start of the generic parameter clause (also <).
+  const FUNCTION_TITLE = {
+    begin: lookahead(/\bfunc\b/),
+    relevance: 0,
+    contains: [
+      {
+        className: 'keyword',
```

Comment:
We should use `keywords` on the parent rule  instead and you don't need a look ahead.

Reply:
This is probably me incorrectly thinking `begin` consumes the token. It doesn't then?
Will `begin/keyword` or `beginKeyword` work when there's a constains rule (`title`) in this case that also matches?

Again, I probably did this to make sure `func` is matched as a keyword, not as an identifier (`title`).

Reply:
`begin` consumes whatever it matches unless you use `excludeBegin`.  But that's not a problem.   You really shoudl try:

```
beginKeywords: `func`
```

This does the match and handles the fact that it's a keyword with single line of code.

Reply:
Changed `FUNCTION_TITLE` to use `beginKeywords`. However, in `FUNCTION` itself, I'm still using the lookahead as I can't consume the keyword there because I need it for `FUNCTION_TITLE`.

Reply:
I finally see what you are doing... you're using "func" to serve as an alternative "look-behind" for TITLE  otherwise you don't have a great way to avoid it double-matching non-titles.  This isn't a problem for many other languages because there are simply no other title like things.  Swift makes everything complicated when `<` can be a title or a template.

I'll give this some more thought.